satellite/console/consoleweb/server.go

```diff
@@ -408,3 +410,80 @@ func (s *Server) Run(ctx context.Context) (err error) {
 func (s *Server) Close() error {
 	return s.server.Close()
 }
+
+// gzipHandler is used to gzip static content to minify resources if browser support such decoding
+func (s *Server) gzipHandler(fn http.Handler) http.Handler {
+	return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		isBrowserDecodesGzip := strings.Contains(request.Header.Get("Accept-Encoding"), "gzip")
+		extension := filepath.Ext(request.RequestURI)
+		// we have gzipped only fonts, js and css bundles
+		isNeededFormatToGzip := stringInSlice(extension, []string{".js", ".ttf", ".css"})
+
+		// in case if old browser doesn't support gzip decoding or if file extension is not recommended to gzip
+		// just return original file
+		if !isBrowserDecodesGzip || !isNeededFormatToGzip {
+			fn.ServeHTTP(writer, request)
+			return
+		}
+
+		filePath, err := s.getFilePathFromRequest(request)
+		if err != nil {
+			s.log.Info(err.Error())
+			fn.ServeHTTP(writer, request)
+			return
+		}
+
+		gzippedFilePath := filePath + ".gz"
+
+		// loading gzipped analog of original file
+		byt, err := ioutil.ReadFile(gzippedFilePath)
```

Comment:
Rather than reading the file into memory, pass it into file handler with the appropriate modified path.